### ISSUE TITLE
Provide better readable error message when build dependency (header) is missing

### ISSUE THREAD

**Issue by sedrubal** [pypa/setuptools#2524]
When you install a source package using pip or whatever package manager and a (C/C++) dependency (header file) is missing during compilation a quite terrible error message appears, which is really hard to read and understand. It would be great if this error message would be better readable.

*In reference to this issue: https://github.com/pypa/pip/issues/9154*

**Comment by jaraco** [comment]
Agreed that would be nice. Unfortunately, the support for compiled modules is unique across platforms (XCode, MSVC, gnu, ...) and is fraught with all of the challenges of distutils and monkeypatching and more. So while I agree it would be nice to improve the error message, I expect any improvements will be particularly difficult until these interfaces can be cleaned up. I welcome contributions.
